src/main/java/com/machine/learning/experimenter/CrossValidator.java

```diff
@@ -5,12 +5,65 @@
 import com.machine.learning.model.DataModel;
 import com.machine.learning.model.Result;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 public class CrossValidator {
-    public CrossValidator(Classifier classifier, DataModel dataModel) {
+    private Classifier classifier;
+    private List<List<List>> folds;
+
+    /**
+     * Performs class validation on all of the data
+     *
+     * @param classifier Algorithm to test
+     * @param dataModel Data to test on
+     * @param numberOfFolds Number of folds to test with
+     */
+    public CrossValidator(Classifier classifier, DataModel dataModel, Integer numberOfFolds) {
+        this.classifier = classifier;
+        this.folds = createFolds(dataModel.getData().orElse(Collections.emptyList()), numberOfFolds);
+
+    }
+
+    /**
+     * Creates the folds to test with
+     *
+     * @param data data to test with
+     * @param numberOfFolds number of folds to create
+     * @return List of folds to test with
+     */
+    public List<List<List>> createFolds(List<List> data, Integer numberOfFolds) {
+        List<List<List>> folds = new ArrayList<>();
+        Collections.shuffle(new ArrayList<>(data));
+        Integer segmentSize = data.size()/numberOfFolds;
+        for (int i = 0; i < numberOfFolds; i++) {
+            folds.add(data.subList(segmentSize*i, segmentSize*(i+1)));
+        }
 
```

Comment:
Doesn't this miss any remaining points, up to `segmentSize - 1` points. Maybe it would be better to give these remaining points to the first `data.size() % segmentSize` folds?